app/services/variant_units/option_value_namer.rb

```diff
@@ -64,36 +64,42 @@ def option_value_value_unit_scaled
     def scale_for_unit_value
       units = {
         'weight' => {
-          1.0 =>         { 'name' => 'g',  'system' => 'metric' },
-          28.35 =>       { 'name' => 'oz', 'system' => 'imperial' },
-          453.6 =>       { 'name' => 'lb', 'system' => 'imperial' },
-          1000.0 =>      { 'name' => 'kg', 'system' => 'metric' },
-          1_000_000.0 => { 'name' => 'T',  'system' => 'metric' }
+          1.0 => { 'name' => 'g', 'system' => 'metric' },
+          28.35 => { 'name' => 'oz', 'system' => 'imperial' },
+          453.6 => { 'name' => 'lb', 'system' => 'imperial' },
+          1000.0 => { 'name' => 'kg', 'system' => 'metric' },
+          1_000_000.0 => { 'name' => 'T', 'system' => 'metric' }
         },
         'volume' => {
-          0.001 =>  { 'name' => 'mL', 'system' => 'metric' },
-          1.0 =>    { 'name' => 'L', 'system' => 'metric' },
+          0.001 => { 'name' => 'mL', 'system' => 'metric' },
+          1.0 => { 'name' => 'L', 'system' => 'metric' },
           1000.0 => { 'name' => 'kL', 'system' => 'metric' }
         }
       }
 
-      # Find the largest available and compatible unit where unit_value comes
-      # to >= 1 when expressed in it.
-      # If there is none available where this is true, use the smallest available unit.
       scales = units[@variant.product.variant_unit]
       product_scale = @variant.product.variant_unit_scale
       product_scale_system = scales[product_scale.to_f]['system']
+
+      largest_unit = find_largest_unit(scales, product_scale_system)
+      [largest_unit[0], largest_unit[1]["name"]]
+    end
+
+    # Find the largest available and compatible unit where unit_value comes
+    #   to >= 1 when expressed in it.
+    # If there is none available where this is true, use the smallest available unit.
+    def find_largest_unit(scales, product_scale_system)
       largest_unit = scales.select { |scale, unit_info|
         unit_info['system'] == product_scale_system &&
           @variant.unit_value / scale >= 1
       }.max
-      largest_unit = units[@variant.product.variant_unit].first if largest_unit.nil?
+      return scales.first if largest_unit.nil?
 
-      [largest_unit[0], largest_unit[1]["name"]]
```

Comment:
Good idea to pull this into a method. I noticed too that doing things like `largest_unit[1]["name"]` isn't great and would definitely like to refactor further, but can do that in another PR.

Reply:
yeah, that's how max works. maybe we can add a comment above this line "max returns an array with two elements"